controllers/halayerset_controller_test.go

```diff
@@ -175,52 +178,104 @@ var _ = Describe("High Availability Layer Set CR", func() {
 		expectedCommands := [][]string{
-			{"pcs", "stonith", "create", originalFenceAgentName, fenceAgentType},
+			{"pcs", "stonith", "create", firstOrgFenceAgent, fenceAgentType},
+		}
+		forbiddenCommands := [][]string{
+			{"pcs", "resource", "remove"},
+			{"pcs", "stonith", "create", updatedFenceAgentName, fenceAgentType},
 		}
 		verifyExpectedCommands(expectedCommands)
+		verifyNoUnExpectedCommand(forbiddenCommands)
 	case deleteCR:
-		Eventually(func() int { return len(execCmdCommands) }, time.Second*10, time.Millisecond*10).Should(BeEquivalentTo(3))
 		expectedCommands := [][]string{
 			{"pcs", "status", "xml"},
-			{"pcs", "stonith", "create", originalFenceAgentName, fenceAgentType},
-			{"pcs", "resource", "remove", originalFenceAgentName, "--force"},
+			{"pcs", "stonith", "create", firstOrgFenceAgent, fenceAgentType},
+			{"pcs", "resource", "remove", firstOrgFenceAgent, "--force"},
+		}
+		forbiddenCommands := [][]string{
+			{"pcs", "resource", "remove", updatedFenceAgentName, "--force"},
+			{"pcs", "stonith", "create", updatedFenceAgentName, fenceAgentType},
 		}
 		verifyExpectedCommands(expectedCommands)
+		verifyNoUnExpectedCommand(forbiddenCommands)
 	case updateCR:
-		Eventually(func() int { return len(execCmdCommands) }, time.Second*10, time.Millisecond*10).Should(BeEquivalentTo(5))
 		expectedCommands := [][]string{
-			{"pcs", "stonith", "create", originalFenceAgentName, fenceAgentType},
+			{"pcs", "stonith", "create", firstOrgFenceAgent, fenceAgentType},
+			{"pcs", "stonith", "create", secondOrgFenceAgent, fenceAgentType},
 			{"pcs", "status", "xml"}, //appears twice
-			{"pcs", "resource", "remove", originalFenceAgentName, "--force"},
+			{"pcs", "resource", "remove", firstOrgFenceAgent, "--force"},
 			{"pcs", "stonith", "create", updatedFenceAgentName, fenceAgentType},
+			{"pcs", "stonith", "update", secondOrgFenceAgent},
 		}
+		forbiddenCommands := [][]string{
+			{"pcs", "resource", "remove", updatedFenceAgentName, "--force"},
+			{"pcs", "resource", "remove", secondOrgFenceAgent, "--force"},
+		}
+		//debug()
 		verifyExpectedCommands(expectedCommands)
+		verifyNoUnExpectedCommand(forbiddenCommands)
 	}
 
 }
 
+func debug() {
+	for i := 0; i < 30; i++ {
+		time.Sleep(time.Second)
+	}
+	fmt.Println("Done")
+}
+
 func verifyExpectedCommands(expectedCommands [][]string) {
-	var actualCommands [][]string
+	Eventually(isExpectedCommandsFound(expectedCommands), time.Second*10, time.Millisecond*10).Should(BeTrue())
+}
+
+func isExpectedCommandsFound(expectedCommands [][]string) bool {
+	for _, expectedCommand := range expectedCommands {
+		if !containsStringSlice(expectedCommand, getActualCommands()) {
+			return false
+		}
+	}
+	return true
+}
+
+func getActualCommands() [][]string {
+
 	for len(execCmdCommands) > 0 {
-		actualCommand := <-execCmdCommands
-		actualCommands = append(actualCommands, actualCommand.command)
+		actualCommands = append(actualCommands, <-execCmdCommands)
```

Comment:
What does `<-execCmdCommands` means? The address of the pointer?

Reply:
`execCmdCommands` is a channel of []string. the `<-` operator takes an element out of the channel

Reply:
1. I see, then eventually `<-execCmdCommands` is one element from the channel of type string? 
2. If `execCmdCommands` has plenty of elements then, `<-execCmdCommands` returns the first one?